api/src/main/java/org/openmrs/api/db/ContextDAO.java

```diff
@@ -62,6 +62,29 @@
 	 */
 	public User getUserByUuid(String uuid) throws ContextAuthenticationException;
 	
+	/**
+	 * Gets a user given the username. Privilege checks are not done here because this is used by the
```

Comment:
Instead of introducing these new methods, can't we just use the Daemon functionality? https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/context/Daemon.java#L28-L29

Reply:
Could we leave this as a further improvement tracked by new ticket? While you're right this was just following the current pattern of the existing `getUserByUuid(String)` that also did not use the Daemon thread (and weirdly reimplemented the same method form `UserDAO` btw...)

Reply:
Since getUserByUuid() is not making any changes to the database, may be yes. But do you still think so even for saveUser()?